groundwork/patterns/gw_recipes_pattern.py

```diff
@@ -60,16 +60,20 @@ def __deactivate_recipes(self, plugin, *args, **kwargs):
         for recipe in recipes.keys():
             self.unregister(recipe)
 
-    def register(self, name, path, description, final_words=None):
+    def register(self, name, path, description, final_words=None,
+                 pre_build_validation=None, post_build_validation=None):
```

Comment:
Do not call it pre_build_validatio, as it may not be used for validation.
Call it  pre_hook & post_hook